### PR TITLE
Clinvar2

### PR DESCRIPTION
**Pull request by northwestwitch**
The same code as the old PR, but rebased to master.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
-- Added an analysis report page (html and PDF format) containing phenotype, gene panels and variants that are relevant to solve a case.
```
Comment (moonso) [inline CHANGELOG.md]: Why is this line not in the rebased version? This line comes from your earlier PR, right? Then it should be included if you have rebased this PR...

Comment (northwestwitch) [reply CHANGELOG.md]: It was included but I removed it at the conflict resolve. I'll put it back!

File: scout/adapter/mongo/clinvar.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+LOG = logging.getLogger(__name__)
+
+class ClinVarHandler(object):
+    """Class to handle clinvar submissions for the mongo adapter"""
+
+    def add_clinvar_submission(self, submission_obj, user, institute, case_id):
+        """Add a clinvar submission for a case. A submission consists of one or more variants added to the clinvar collection.
+
+            Args:
+                submission_obj(str)
```
Comment (moonso) [inline scout/adapter/mongo/clinvar.py]: The submission object is not a string, please change and describe more.
Is is just a dictionary with variants or is there any extra information?

File: scout/adapter/mongo/clinvar.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                user(str)
+                institute(str)
+                case_id(str)
+
+            Returns:
```
Comment (moonso) [inline scout/adapter/mongo/clinvar.py]: It looks like it can also return 0 and -1, what does that mean?

File: scout/parse/clinvar.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                    csv_line.append('""') # empty cell
+
+        if csv_line not in csv_lines and len(csv_line)>0:
+            csv_lines.append(csv_line)
+
+    return csv_lines
+
+
+def extract_submission_csv_lines(clinvars_dictlist):
+    """Parses a list of clinvar submission objects (variants) and creates the lines for printing
```
Comment (moonso) [inline scout/parse/clinvar.py]: There are great docstrings above! Could you add the same information here?

File: scout/parse/clinvar.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                temp_cd_fields = []
+                for cd_field in casedata_header:
+                    temp_cd_fields.append('"'+case[cd_field.replace(' ','_')]+'"')
+                casedata_lines.append(','.join(temp_cd_fields))
+
+    return '"'+'","'.join(variants_header)+'"', '"'+'","'.join(casedata_header)+'"', clinvar_lines, casedata_lines
+
+
+def create_clinvar_submission_dict(variant_header, variant_lines, casedata_header, casedata_lines, variant_types):
+    """
```
Comment (moonso) [inline scout/parse/clinvar.py]: Also nice with a lot of information here, please add `Args` and `Returns`.
I mean you could still keep this detailed information and add Args and Returns afterwards. It is nice if all docstrings has the same structure

File: scout/server/blueprints/variants/controllers.py [hunk 6]
```diff
@@ -732,6 +741,35 @@ def cancer_variants(store, request_args, institute_id, case_name):
     )
     return data
 
+def clinvar_export(store, institute_id, case_name, variant_id):
+
```
Comment (moonso) [inline scout/server/blueprints/variants/controllers.py]: What about docstrings here? Could you add that as well?

File: tests/adapter/mongo/test_clinvar_handler.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
+import logging
+import pymongo
+from numbers import Number
+
+LOG = logging.getLogger(__name__)
+
+def get_test_submission_object():
```
Comment (moonso) [inline tests/adapter/mongo/test_clinvar_handler.py]: Great work on the tests 🌮

**Comment by northwestwitch** [comment]
I've fixed the code according your comments @moonso. OMG the comments in the mongo adapter were sooo bad!! :)

**Pull request merged**